### ISSUE TITLE
Maya: Render submissions with Arnold do not adhere to 'aovSeparator'

### ISSUE THREAD

**Issue by BigRoy** [pypeclub/OpenPype#2636]
**Describe the bug**

I'm hitting a bug where Arnold renders can't be published due to the filenames in the `metadata.json` having the wrong AOV separator in the filename compared to what is being rendered. The scene submission to Deadline validated and submitted fine.

Did we do something wrong for this to happen in our case?

The [`lib_renderproducts`](https://github.com/pypeclub/OpenPype/blob/13266d005c482d705defd5c233259813115ba829/openpype/hosts/maya/api/lib_renderproducts.py) does correctly return the output filenames when used like this:
```python
import openpype.hosts.maya.api.lib_renderproducts as rp

# Get the file list for the master layer
renderer = rp.get("mylayer")
for product in renderer.get_render_products():
    print(renderer.get_files(product))
```
However, the [Collect Render logic overrides the separator](https://github.com/pypeclub/OpenPype/blob/13266d005c482d705defd5c233259813115ba829/openpype/hosts/maya/plugins/publish/collect_render.py#L176-L186) using the OpenPype settings which is basically [making `lib_renderproducts` use that instead](https://github.com/pypeclub/OpenPype/blob/13266d005c482d705defd5c233259813115ba829/openpype/hosts/maya/api/lib_renderproducts.py#L183) and thus NOT returning the aov separator the renderer is actually using. 

Which is unfortunate, because that's exactly the information `lib_renderproducts` should provide you. Without prejudice it should tell you that "if you render this layer with this renderer, these are your output files." and should have no 'custom overrides' deviating from what the renderer would actually be doing. I'd opt instead to have a validator that validates whether the output matches whatever setting you might've set - plus potentially providing a "repair" option for that specific renderer.

It currently passing into rendering and being unable to publish basically means the AOV separator defined in settings is not adhered to in the render output and thus rendered images from Maya Arnold can't publish AOVs when the separator is using `_` (which according to the Collect Render logic is also the default fallback?) unless you first explicitly set it so.

[cuID:OP-2577]

**Comment by BigRoy** [comment]
I think the issue is that `CreateRender` tries to set the values for the current renderer upon creating the render instance and it didn't for Arnold in this case. However this apparently is problematic.

We tend to mix renderers even in different renderlayers and thus not all render settings are correctly set for each renderer. Nor are the settings validated on submission.

**Comment by antirotor** [comment]
This is somewhat related to this #2640. Unfortunately I can't imagine this feature would work in scenario where you are using different renderers per render layer. But it should coexist peacefully. AOV separator should be set only by creator and validated by render settings validator. It is now forced in collector as a quick patch for clients as inconsistency of AOV separator character was production breaking for them.

I think #2640 should by default set only stuff that is absolutely needed in the rest of the code and it should be tighly bound with render settings validator. This would allow you to leave it almost empty to do whatever anarchic dark magic you'd like but also allow you to preset more strict workflows

**Comment by BigRoy** [comment]
> I think https://github.com/pypeclub/OpenPype/pull/2640 should by default set only stuff that is absolutely needed in the rest of the code and it should be tighly bound with render settings validator. This would allow you to leave it almost empty to do whatever anarchic dark magic you'd like but also allow you to preset more strict workflows

Thanks. I will set up a draft PR where I start pushing away at changes to slowly work towards something.

**Comment by antirotor** [comment]
This will be solved by #3097
